### DISCUSSION TITLE
Setting roles based on login type

### DISCUSSION THREAD

**Discussion by praj18** [supabase/supabase#4774] [Questions]
I want to set roles based on login type. If a user has logged in using social auth such as google, github, etc., I want to set that user's role as 'visitor', whereas if the user uses email and password to sign up, I want to set the role to be 'customer'.

Additionally, if a user has previously signed up with email and password, and now uses social auth to login, I do not want the roles to switch from 'customer' to 'visitor'.

Is this possible? If yes, how should I go about doing it?

**Comment by GaryAustin1** [comment]
If you have a trigger function on auth.users table for insert, update you should be able to check if the encrypted_password column has a value and set a user_role table to customer else to visitor for that uuid.   I don't use 3rd party login though, so you need to verify the password column is empty string/null for 3rd party only login.  I ASSUME it is as no password is involved on the supabase side.

**Comment by jacobgad** [comment]
Ok so assuming i have 3 tables, 
1. auth.users
2. public.profiles
3. public.user_roles

what is the recomended way to join these tables together?
how would i write a row level securety policy that checks if the requesting user is an admin by looking it up in the user_roles table?

I hope that makes sense,
Thank you

**Reply by GaryAustin1** [reply]
Maybe this will point you in the right direction:
https://supabase.com/docs/guides/auth/row-level-security#policies-with-joins

**Comment by MarcosRZ** [comment]
How about using [custom claims](https://github.com/supabase-community/supabase-custom-claims) ?

You can add almost everything (jsonb support) to your users using `set_claims(...)` function and access this information via `get_claims(...)` in RLS Policies, auth middleware, etc.

I'm currently using this approach to add an 'admin' claim to some specific users.
